File: client/lib/features/onboarding/models/allergen_tag.dart
class AllergenTag {
  final int id;
  final String label;
  final String? emoji;
  final String? description;
  final String? category;

  AllergenTag({
    required this.id,
    required this.label,
    this.emoji,
    this.description,
    this.category,
  });

  factory AllergenTag.fromJson(Map<String, dynamic> json) {
    return AllergenTag(
      id: json['id'],
      label: json['label'],
      emoji: json['emoji'],
      description: json['description'],
      category: json['category'],
    );
  }
}

File: server/src/main/java/cn/cuckoox/wisediet/model/AllergenTag.java
package cn.cuckoox.wisediet.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Table("allergen_tags")
public class AllergenTag {
    @Id
    private Long id;
    private String label;
    private String emoji;
    private String description;
    private String category;
}
